interfaces/seccomp/backend.go

```diff
@@ -174,14 +178,13 @@ func (b *Backend) Setup(snapInfo *snap.Info, opts interfaces.ConfinementOptions,
 		return fmt.Errorf("cannot synchronize security files for snap %q: %s", snapName, err)
 	}
 
+	compiler := seccomp_compiler.NewAtPath(seccompToBpfPath())
```

Comment:
it's a bit unclear to me why the way to find the compiler would remain here, seems a strange layering. For easier mocking in tests?  we should probably support better mocking in the new package then

Reply:
I didn't want to bake in the assumption that `snap-seccomp` is at the same location as the current binary into the package. Felt like the caller would be better suited for locating the compiler.

Reply:
Ok, two considerations

* given we are touching this we can address the FIXME in that function and switch to use cmd.InternalToolPath

* I have a small preference for the API to be  New(lookupTool func(name string) string) but consider that a nitpick,  so it would become  seccomp_compiler.New(cmd.InternalToolPath)

Reply:
Or use the signature of https://golang.org/pkg/os/exec/#LookPath for the constructor argument.

Reply:
I'll update the API.

> cmd.InternalToolPath

That helper has the same problem operating in a bootstrap environment as the previous one had.

Reply:
The API has been updated, added some tests for that too.